.cirrus.yml

```diff
@@ -0,0 +1,121 @@
+common_steps_template: &COMMON_STEPS_TEMPLATE
+  install_prerequisites_script: |
+    set -uexo pipefail
+    if [ "$OS_NAME" == "linux" ]; then
+      packages="git-core make g++ gdb curl libcurl3 tzdata zip unzip xz-utils"
+      if [ "$MODEL" == "32" ]; then
+        dpkg --add-architecture i386
+        packages="$packages g++-multilib libcurl3-gnutls:i386"
+      fi
+      if [ "${DMD:0:4}" == "gdmd" ]; then
+        packages="$packages sudo software-properties-common wget"
+      fi
+      apt-get -q update
+      apt-get install -yq $packages
+    elif [ "$OS_NAME" == "darwin" ]; then
+      # required for install.sh
+      brew install gnupg
+    elif [ "$OS_NAME" == "freebsd" ]; then
+      pkg install -y git gmake bash
+      rm /usr/bin/make
+      ln -s /usr/local/bin/gmake /usr/bin/make
+    fi
+    # create a `dmd` symlink to the repo dir, necessary for druntime/Phobos
+    ln -s $CIRRUS_WORKING_DIR ../dmd
+  install_host_compiler_script: |
+    source ci.sh
+    # kludge
+    if [ "${DMD:0:4}" == "gdmd" ]; then export DMD="gdmd"; fi
+    install_d "$DMD"
+  setup_repos_script: |
+    export BRANCH=${CIRRUS_BASE_BRANCH:-$CIRRUS_BRANCH}
+    source ci.sh
+    setup_repos
+  build_script: |
+    source ci.sh
+    build
+  test_dmd_script: |
+    if [ -z "$CIRRUS_PR" ]; then export FULL_BUILD="true"; fi
```

Comment:
Considering that it is so fast, we could probably set `FULL_BUILD` to `true` all the time.

Reply:
Yeah it probably won't make much of a difference. But `ci.sh` only really uses it for Linux at the moment.